apps/api/src/integration/integration.service.ts

```diff
@@ -729,4 +627,174 @@ export class IntegrationService {
       )
     }
   }
+
+  /**
+   * Tests a new integration in the given workspace. The user needs to have
+   * `CREATE_INTEGRATION` and `READ_WORKSPACE` authority in the workspace.
+   *
+   * Check `validateIntegrationCreation` for more details regarding validation.
+   *
+   * @param user The user creating the integration
+   * @param dto The integration data
+   * @param workspaceSlug The slug of the workspace the integration is being
+   * created in
+   * @returns { success: true } if the integration configuration is valid
+   */
+  async testIntegration(
+    user: AuthenticatedUser,
+    dto: CreateIntegration,
+    workspaceSlug: Workspace['slug'],
+    privateKey?: Project['privateKey']
+  ) {
+    this.logger.log(
+      `User ${user.id} is testing configuration of integration ${dto.name} in workspace ${workspaceSlug}`
+    )
+
+    await this.validateIntegrationCreation(user, dto, workspaceSlug, privateKey)
+
+    return { success: true }
+  }
+
+  /**
+   * If the integration is of type `PROJECT`, the user needs to have `READ_PROJECT`
+   * authority in the project specified by `projectSlug`.
+   *
+   * If the integration is of type `ENVIRONMENT`, the user needs to have `READ_ENVIRONMENT`
+   * authority in the environment specified by `environmentSlug`.
+   *
+   * If the integration is of type `PROJECT` and `environmentSlug` is provided,
+   * the user needs to have `READ_ENVIRONMENT` authority in the environment specified
+   * by `environmentSlug`.
+   *
+   * The integration configuration is then verified through the `validateConfiguration`
+   * method of the integration plugin.
+   *
+   * @param user The user creating the integration
+   * @param dto The integration data
+   * @param workspaceSlug The slug of the workspace the integration is being
+   * created in
+   * @returns environments, project, workspaceId
+   */
+  private async validateIntegrationCreation(

```

Comment:
we should pass an override in here - `onlyValidateMetadata?: boolean`. This would instruct the function to validate only the metadata of the integration. This would be set to true from `validateIntegrationMetadata`

Reply:
Your concerns are valid, but here's the case:
- We know the user will have access to the workspace, and the project (if present) because we check for permissions in `authorizeAccessOverIntegratios`
- A new feature introduced yesterday removes the ability to update project for an integration. So this only leaves us with environments, and metadata specific configs.